dips/dip-178.md

```diff
@@ -0,0 +1,292 @@
+* "Merchant VASP" - The VASP representing the merchant. In a refund scenario, this is the VASP that will send the funds and eventually submit the on-chain transaction to the blockchain
+* "Customer" - The person who wishes to pay the merchant for goods or services
+
+
+---
+# P2M Refund Off-Chain Protocol Specification
+This section specifies the required structures, payloads and expected sequence of calls based on this protocol.
+
+Content:
+* [High Level Flow](#-high-level-flow)
+* [Interface](#-interface)
+* [Refund Sequences](#-refund-sequences)
+* [Request / Response Payload](#-request--response-payload)
+* [Structures](#-structures)
+
+---
+## High-Level Flow
+---
+The high-level flow of a refund payment is:
+1. A refund flow starts with either (a) The merchant transmits a refund request with the required details to its VASP or (b) the merchant VASP itself triggers a refund payment     
+2. The merchant VASP will init a refund payment and send the details to the customer VASP (probably after performing some basic validations)
+3. The customer VASP will verify the refund details provided, and if all is in order, will indicate it is ready to accept the refund. **Note** that if the refund amount exceeds the amount that requires dual attestation signature, the customer VASP must provide the signature as part ot its response, so the merchant VASP can later inlcude it in the on-chain transaction 
+4. The merchant VASP submits a transaction on-chain and in parallel notify the customer VASP the refund is ready for settlement. Final validation will occur before this step (for example, verify the requested refund amount is not greater than the amount of original payment)
+
+---
+## Interface
+---
+This section describes a set of functionalities / activities that should be made available by the different VASPs in order to support P2M Refund flow based on this DIP. Each functionality can be conceptually compared to an API endpoint. Similar to API documentation, it includes the expected input and output for each functionality.
+
+## Command Types
+The following is a list of new and existing values for the `command_type` field of the `CommandRequestObject`. Each command type denotes a step in the refund payment off-chain negotiations and specifies the data of the requests and the response 
+* `InitRefundPayment` - Initializes a refund payment
+* `RefundReadyForSettlementNotification` - Notifies the payment is ready for settlement on-chain
+* `AbortRefund` - Aborts a payment
+
+| Command Type |Direction| Description | Request Data | Response Data |
+|-|-|-|-|-|
+|[InitRefundPayment](#initrefundpayment)|Merchant VASP > Customer VASP|This functionality allows the merchant VASP to init the refund. For example, this may occur after the merchant requested its VASP to refund the customer due to some items being out of stock|Reference ID ; Refund ID ; (optional) Amount|(optional) Recipient Signature|
```

Comment:
Why do we need a Refund ID in addition to the Reference ID?

If we do need it, can we include a snippet explaining its purpose?

Reply:
There could be multiple refunds for the same payment.
In order to identify a specific refund and maintain a successive protocol, refund ID must be passed